### PR TITLE
Fix backwards compatibility with block version 1

### PR DESCRIPTION
**Pull request by simonswine**
#767 broke compatibility with version 1 blocks, this fixes the two issues and also adds sample blocks and tests to ensure compatibility.

### CODE REVIEW

**Comment by cyriltovena** [comment]
Did you figure why it broke ?

File: pkg/phlaredb/block_querier.go [hunk 1]
```diff
@@ -983,17 +986,29 @@ func (b *singleBlockQuerier) SelectMatchingProfiles(ctx context.Context, params
...
-	pIt := query.NewJoinIterator(
-		0,
-		[]query.Iterator{
+
+	var (
+		buf       [][]parquet.Value
+		joinIters []query.Iterator
+	)
+
+	if b.meta.Version >= 2 {
```
Comment (cyriltovena) [inline pkg/phlaredb/block_querier.go:995]: I think I want to include total sample in v2 WDYT ?

**Review by cyriltovena — APPROVED** [review]
LGTM

File: pkg/phlaredb/block_querier.go (hunk 1 above)
Comment (simonswine) [reply pkg/phlaredb/block_querier.go:995]: I would really throw around with versions and make that version+1, also the sorted should be a version+1

File: pkg/phlaredb/query/repeated.go [hunk 2]
```diff
@@ -83,8 +84,10 @@ func (it *repeatedPageIterator[T]) seekRowNum() int64 {
 		return i.RowNumber()
 	case int64:
 		return i
+	case uint32:
+		return int64(i)
```
Comment (simonswine) [inline pkg/phlaredb/query/repeated.go:88]: Problem #2, panic with uint32s

File: pkg/phlaredb/block_querier.go [hunk 3]
```diff
@@ -983,17 +986,29 @@ func (b *singleBlockQuerier) SelectMatchingProfiles(ctx context.Context, params
...
+		joinIters = []query.Iterator{
 			b.profiles.columnIter(ctx, "SeriesIndex", newMapPredicate(lblsPerRef), "SeriesIndex"),
 			b.profiles.columnIter(ctx, "TimeNanos", query.NewIntBetweenPredicate(model.Time(params.Start).UnixNano(), model.Time(params.End).UnixNano()), "TimeNanos"),
 			b.profiles.columnIter(ctx, "StacktracePartition", nil, "StacktracePartition"),
```
Comment (simonswine) [inline pkg/phlaredb/block_querier.go:999]: Problem #1: This requires the column to be there, maybe should have a orNull predicate or something.

**Pull request merged**